modelon/impact/client/sal/project.py

```diff
@@ -43,3 +59,45 @@ def project_content_upload(
                 'options': json.dumps(options),
             }
             return self._http_client.post_json(url, files=multipart_form_data)
+
+    def fmu_upload(
+        self,
+        workspace_id: str,
+        project_id: str,
+        content_id: str,
+        fmu_path: str,
+        library: str,
+        class_name: Optional[str] = None,
+        overwrite: bool = False,
+        include_patterns: Optional[Union[str, List[str]]] = None,
+        exclude_patterns: Optional[Union[str, List[str]]] = None,
+        top_level_inputs: Optional[Union[str, List[str]]] = None,
+        step_size: float = 0.0,
+    ):
+        url = (
+            self._base_uri
+            / f"api/workspaces/{workspace_id}/projects/{project_id}/content/"
+            f"{content_id}/models"
+        ).resolve()
+        default_class_name = ".".join(
+            [library, os.path.split(fmu_path)[-1].strip('.fmu')]
+        )
```

Comment:
`library` should not be sent down all the way down here to help with default value. It should be done higher up and it should use `content.relpath` to help find a default. `class_name` should probably not be optional here.